Framework/DataHandling/src/SaveAscii2.cpp

```diff
@@ -499,5 +508,45 @@ void SaveAscii2::writeTableWorkspace(const ITableWorkspace_const_sptr &tws, cons
   file.close();
 }
 
+/**
+ * Retrieves sample log value and its unit. In case they are not defined they
+ * are replaced with 'not defined' meassage and empty string, respectively.
+ * @param logName :: The user-defined identifier for sample log
+ * @return A pair of strings containing sample log value and its unit
+ */
+std::pair<std::string, std::string> SaveAscii2::sampleLogValueUnit(const std::string &logName) {
+  auto run = m_ws->run();
+  // Gets the sample log value
+  std::string sampleLogValue = "";
+  try {
+    sampleLogValue = boost::lexical_cast<std::string>(run.getLogData(logName)->value());
+  } catch (Exception::NotFoundError &) {
+    g_log.warning("Log " + logName + " not found.");
+    sampleLogValue = "Not defined";
+  }
+  // Gets the sample log unit
+  std::string sampleLogUnit = "";
+  try {
+    sampleLogUnit = boost::lexical_cast<std::string>(run.getLogData(logName)->units());
+  } catch (Exception::NotFoundError &) {
+    sampleLogUnit = "";
+  }
+  return std::pair(sampleLogValue, sampleLogUnit);
+}
+
+/**
+ * Writes the file header containing the user-defined sample logs.
+ * @param logList :: A vector of strings containing user-defined identifiers for
+ * sample logs
+ * @param outputFile :: A reference to the output stream
+ */
+void SaveAscii2::writeFileHeader(const std::vector<std::string> &logList, std::ofstream &outputFile) {
+  for (const auto &logName : logList) {
+    const std::pair<std::string, std::string> readLog = sampleLogValueUnit(logName);
+    outputFile << logName << m_sep << readLog.first << m_sep << readLog.second << '\n';
```

Comment:
Is it worth managing commas in the sample log values? One of the log entries called experiment_title looks like it might have commas in (although doesn't on the examples I've looked at).
You could add surrounding quotes if it contains a comma or in one of your other PRs I remember you converted to ;

Reply:
This indeed may be an issue for CSV files, then a solution could be to use a boost library to sanitize the string just like in GenerateLogbook.py algorithm.